tasks/mysql_secure_installation.yml

```diff
@@ -0,0 +1,63 @@
+---
+
+# supported for ansible ver => 2.0
+#- name: Install python-mysqldb for Ansible
+#  package: pkg=python-mysqldb state=present
+
+
+- name: Install python-mysqldb for Ansible
+  apt: name=python-mysqldb state=present
+  when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
+
+- name: Install python-mysqldb for Ansible
+  yum: name=MySQL-python state=present
+  when: ansible_os_family == 'RedHat' or ansible_os_family == 'Oracle Linux'
+
+- name: root password is present
+  mysql_user: name=root host={{ item }} password={{ mysql_root_password }} state=present
+  with_items:
+    - '::1'
+    - '127.0.0.1'
+    - 'localhost'
+
+- name: root has .my.cnf
+  template: src=my.cnf.j2 dest=/root/.my.cnf 
```

Comment:
> Also the .my.cnf has to be copied before the "root password is present" task, else the login will fail.

Ok, let's discuss this. I'm not agree. It will fail if I'll rearrange tasks in case user has no root password set up. Because after installing _.my.cnf_ with credentials, next task will use those and fail, because it has no credentials. Using following sequence it doesn't fail, it's proved by dozens of mysql installations, and I also performed one more and test to check it again